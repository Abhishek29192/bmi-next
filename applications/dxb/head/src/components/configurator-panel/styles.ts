import { styled } from "@mui/material/styles";
import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  accordionSummaryClasses
} from "@mui/material";

const PREFIX = "configuratorPanelStyles";
export const classes = {
  selectedTitle: `${PREFIX}-selectedTitle`,
  title: `${PREFIX}-title`,
  summary: `${PREFIX}-summary`
};

export const Root = styled("div")(({ theme }) => ({
  "& + &": {
    marginTop: "16px"
  }
}));

export const StyledAccordion = styled(Accordion)(({ theme }) => ({
  backgroundColor: theme.colours.alabaster,
  [`&.Mui-disabled`]: {
    backgroundColor: theme.colours.alabaster
  }
}));

export const StyledAccordionSummary = styled(AccordionSummary)(({ theme }) => ({
  paddingLeft: "32px",
  paddingRight: "42px",
  color: theme.colours.slate,
  opacity: 1,
  "&:hover": {
    color: theme.colours.charcoal,
    [`& .${classes.title}`]: {
      [`& > .${classes.selectedTitle}`]: {
        color: theme.colours.focus
      }
    }
  },
  [`& .${classes.title}`]: {
    [`& > .${classes.selectedTitle}`]: {
      color: theme.colours.inter,
      [theme.breakpoints!.up!("sm")]: {
        fontSize: "1.6875rem"
      }
    }
  },
  [`&.Mui-disabled`]: {
    opacity: 1,
    [`& .${accordionSummaryClasses.expandIconWrapper}`]: {
      display: "none"
    }
  },
  [`& .${accordionSummaryClasses.content}`]: {
    marginTop: "32px",
    marginBottom: "32px"
  },
  [`& .${accordionSummaryClasses.expandIconWrapper}`]: {
    fontSize: "1.33rem"
  }
}));

export const StyledAccordionDetails = styled(AccordionDetails)(({ theme }) => ({
  backgroundColor: theme.colours.alabaster,
  padding: "0 32px 32px",
  flexFlow: "column"
}));

export const Content = styled("div")({
  marginBottom: "20px"
});
