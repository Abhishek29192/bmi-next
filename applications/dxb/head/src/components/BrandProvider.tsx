import React, { createContext, useLayoutEffect, useRef, useState } from "react";
import classnames from "classnames";
import { ThemeProvider } from "@bmi/components";
import { DialogClassNameContext } from "@bmi/components";
import type { ThemeOptions } from "@material-ui/core/styles";

import styles from "./styles/BrandProvider.module.scss";

export const BrandClassNameContext = createContext<string | null>(null);

const BRANDS_CLASSES: { [key: string]: string | undefined } = {
  AeroDek: styles["AeroDeck"],
  Awak: styles["AeroDeck"],
  Braas: styles["BrandRed"],
  Bramac: styles["BrandRed"],
  Coverland: styles["BrandRed"],
  Esha: styles["BrandGreen"],
  Everguard: styles["Purple"],
  Monarflex: styles["RoyalBlue"],
  Monarplan: styles["Purple"],
  Monier: styles["BrandRed"],
  Necoflex: styles["NavyBlue"],
  Ormax: styles["BrandRed"],
  Redland: styles["BrandRed"],
  Sealoflex: styles["NavyBlue"],
  Sunscape: styles["BrandRed"],
  Vedag: styles["NavyBlue"],
  Villas: styles["RoyalBlue"],
  Wierer: styles["BrandRed"],
  Wolfin: styles["BrandGreen"],
  Zanda: styles["BrandRed"]
};

export const getBrandClassName = (brand?: string): string => {
  return BRANDS_CLASSES[`${brand}`] || "";
};

type NoBrandProps = {
  children: React.ReactNode;
};

type BrandProviderProps = {
  children: React.ReactNode;
  brand?: string;
};

const NoBrandComponent = ({ children }: NoBrandProps) => {
  return (
    <div className={""} data-testid="brand-colors-provider">
      {children}
    </div>
  );
};

const BrandProvider = ({ brand, children }: BrandProviderProps) => {
  if (!brand) {
    return <NoBrandComponent>{children}</NoBrandComponent>;
  }
  const ref = useRef<HTMLDivElement | null>(null);

  const [interColor, setInterColor] = useState<string | null>(null);
  const [focusColor, setFocusColor] = useState<string | undefined>();

  const className = getBrandClassName(brand);

  useLayoutEffect(() => {
    if (ref.current) {
      const style = getComputedStyle(ref.current);
      const interColor = style.getPropertyValue("--color-theme-inter").trim();
      const focusColor = style.getPropertyValue("--color-theme-focus").trim();
      if (interColor) {
        setInterColor(interColor);
      }
      if (focusColor) {
        setFocusColor(focusColor);
      }
    }
  }, []);

  const modifyThemePrimaryColor = (theme: ThemeOptions) => {
    if (interColor) {
      theme.palette = theme.palette || {};
      theme.palette.primary = { main: interColor, dark: focusColor };
    }
    return theme;
  };

  const featureToggle = process.env.GATSBY_ENABLE_BRAND_PROVIDER as
    | boolean // styleguidist's webpack returns boolean value
    | string
    | undefined;

  return featureToggle === "true" || featureToggle === true ? (
    <div
      ref={ref}
      className={classnames(className)}
      data-testid="brand-colors-provider"
    >
      <BrandClassNameContext.Provider value={className}>
        <ThemeProvider modifyTheme={modifyThemePrimaryColor}>
          <DialogClassNameContext.Provider value={className}>
            {children}
          </DialogClassNameContext.Provider>
        </ThemeProvider>
      </BrandClassNameContext.Provider>
    </div>
  ) : (
    <NoBrandComponent>{children}</NoBrandComponent>
  );
};

export default BrandProvider;
