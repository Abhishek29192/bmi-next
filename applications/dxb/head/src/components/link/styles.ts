import AnchorLink from "@bmi-digital/components/anchor-link";
import Button from "@bmi-digital/components/button";
import { styled } from "@mui/material/styles";

const PREFIX = "Link";

export const classes = {
  branded: `${PREFIX}-hasBrand`
};

export const StyledAnchorLink = styled(AnchorLink)(({ theme }) => ({
  [`&.${classes.branded}`]: {
    background: theme.colours.inter,
    borderColor: theme.colours.inter,
    color: theme.colours.white
  }
}));

export const StyledButton = styled(Button)(({ theme }) => ({
  [`&.${classes.branded}`]: {
    background: theme.colours.inter,
    borderColor: theme.colours.inter,
    color: theme.colours.white
  }
}));

export const StyledLinkDialog = styled("div")(({ theme }) => ({
  marginBottom: "2rem",
  maxHeight: "100%",
  overflow: "auto",
  padding: "0 2rem"
}));
