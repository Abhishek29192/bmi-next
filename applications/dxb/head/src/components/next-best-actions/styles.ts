import ArrowForward from "@bmi-digital/components/icon/ArrowForward";
import { styled } from "@mui/material/styles";

export const CtaWrapper = styled("div")({
  fontSize: "1rem",
  display: "inline-flex",
  alignItems: "center"
});

export const StyledArrowForward = styled(ArrowForward)({
  width: "16px",
  marginLeft: "12px",
  marginRight: "12px"
});
