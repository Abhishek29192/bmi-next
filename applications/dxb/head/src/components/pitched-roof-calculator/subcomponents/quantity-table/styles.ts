import { styled } from "@mui/material/styles";

const PREFIX = "quantityTableStyles";
export const classes = {
  smallCell: `${PREFIX}-smallCell`,
  largeCell: `${PREFIX}-largeCell`,
  header: `${PREFIX}-header`,
  mediumHeaderCell: `${PREFIX}-mediumHeaderCell`,
  largeHeaderFirstCell: `${PREFIX}-largeHeaderFirstCell`,
  mediumHeaderFirstCell: `${PREFIX}-mediumHeaderFirstCell`,
  mediumCell: `${PREFIX}-mediumCell`,
  mediumTableRow: `${PREFIX}-mediumTableRow`,
  largeDescription: `${PREFIX}-largeDescription`,
  greyBack: `${PREFIX}-greyBack`,
  smallDescription: `${PREFIX}-smallDescription`,
  packSizes: `${PREFIX}-packSizes`,
  cellMediumRow: `${PREFIX}-cellMediumRow`,
  cellRow: `${PREFIX}-cellRow`,
  cellRowLast: `${PREFIX}-cellRowLast`,
  icon: `${PREFIX}-icon`,
  picture: `${PREFIX}-picture`,
  boldText: `${PREFIX}-boldText`,
  iteratorCellSmall: `${PREFIX}-iteratorCellSmall`,
  iteratorCellMedium: `${PREFIX}-iteratorCellMedium`,
  iteratorCellLarge: `${PREFIX}-iteratorCellLarge`,
  mediumCellBasketIconWrapper: `${PREFIX}-mediumCellBasketIconWrapper`
};

export const StyledQuantityTable = styled("div")(({ theme }) => ({
  [`& .${classes.header}`]: {
    minWidth: "100px",
    textAlign: "center"
  },
  [`& .${classes.greyBack}`]: {
    backgroundColor: "#f7f7f7"
  },
  [`& .${classes.picture}`]: {
    height: "80px",
    width: "80px",
    mixBlendMode: "multiply"
  },

  [`& .${classes.boldText}`]: {
    marginTop: "3px",
    marginLeft: "5px",
    fontWeight: "bold"
  },

  [`& .${classes.icon}`]: {
    fill: theme.colours.inter,
    cursor: "pointer"
  },
  [`& .${classes.largeDescription}`]: {
    marginLeft: "10px",
    alignSelf: "center",
    wordBreak: "break-all"
  }
}));

export const StyledQuantityTableLarge = styled("div")(({ theme }) => ({
  display: "contents",
  [`& .${classes.largeHeaderFirstCell}`]: {
    minWidth: "220px"
  },
  [`& .${classes.header}`]: {
    minWidth: "100px",
    textAlign: "center"
  },
  [`& .${classes.greyBack}`]: {
    backgroundColor: "#f7f7f7"
  },
  [`& .${classes.smallCell}`]: { width: "10%" },
  [`& .${classes.cellRow}`]: {
    display: "flex",
    flexDirection: "row",
    marginBottom: "10px"
  },
  [`& .${classes.picture}`]: {
    height: "80px",
    width: "80px",
    mixBlendMode: "multiply"
  },
  [`& .${classes.boldText}`]: {
    marginTop: "3px",
    marginLeft: "5px",
    fontWeight: "bold"
  },

  [`& .${classes.iteratorCellSmall}`]: {
    maxWidth: "170px",
    "& input": {
      padding: "18px 12px",
      textAlign: "center"
    }
  },
  [`& .${classes.icon}`]: {
    fill: theme.colours.inter,
    cursor: "pointer"
  },
  [`& .${classes.mediumTableRow}`]: {
    verticalAlign: "top",
    "& td": {
      height: "100%"
    },
    "& > td:not(:first-of-type)": {
      padding: "45px 0 32px"
    }
  },
  [`& .${classes.largeDescription}`]: {
    marginLeft: "10px",
    alignSelf: "center",
    wordBreak: "break-all"
  },
  [`& .${classes.largeCell}`]: {
    textAlign: "center",
    width: "17.5%"
  },
  [`& .${classes.iteratorCellLarge}`]: {
    minWidth: "170px",
    "& input": {
      padding: "18px 12px",
      textAlign: "center"
    }
  }
}));

export const StyledQuantityTableMedium = styled("div")(({ theme }) => ({
  display: "contents",
  [`& .${classes.mediumHeaderFirstCell}`]: {
    minWidth: "204px"
  },
  [`& .${classes.mediumHeaderCell}`]: {
    width: "138px",
    textAlign: "center"
  },
  [`& .${classes.greyBack}`]: {
    backgroundColor: "#f7f7f7"
  },
  [`& .${classes.cellRow}`]: {
    display: "flex",
    flexDirection: "row",
    marginBottom: "10px"
  },
  [`& .${classes.picture}`]: {
    height: "80px",
    width: "80px",
    mixBlendMode: "multiply"
  },
  [`& .${classes.boldText}`]: {
    marginTop: "3px",
    marginLeft: "5px",
    fontWeight: "bold"
  },

  [`& .${classes.icon}`]: {
    fill: theme.colours.inter,
    cursor: "pointer"
  },
  [`& .${classes.mediumTableRow}`]: {
    verticalAlign: "top",
    "& td": {
      height: "100%"
    },
    "& > td:not(:first-of-type)": {
      padding: "45px 0 32px"
    }
  },
  [`& .${classes.iteratorCellMedium}`]: {
    maxWidth: "170px",
    marginTop: "22px",
    "& input": {
      padding: "18px 12px",
      textAlign: "center"
    }
  },
  [`& .${classes.mediumCell}`]: {
    marginLeft: "10px",
    textAlign: "center"
  },
  [`& .${classes.mediumCellBasketIconWrapper}`]: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    alignItems: "center"
  },
  [`& .${classes.cellMediumRow}`]: {
    display: "flex",
    flexDirection: "row",
    marginBottom: "10px",
    minWidth: "30%"
  },
  [`& .${classes.largeDescription}`]: {
    marginLeft: "10px",
    alignSelf: "center",
    wordBreak: "break-all"
  }
}));

export const StyledQuantityTableSmall = styled("div")(({ theme }) => ({
  [`& .${classes.smallCell}`]: { width: "10%" },
  [`& .${classes.cellRow}`]: {
    display: "flex",
    flexDirection: "row",
    marginBottom: "10px"
  },
  [`& .${classes.picture}`]: {
    height: "80px",
    width: "80px",
    mixBlendMode: "multiply"
  },
  [`& .${classes.smallDescription}`]: {
    wordBreak: "break-all",
    margin: "10px"
  },
  [`& .${classes.boldText}`]: {
    marginTop: "3px",
    marginLeft: "5px",
    fontWeight: "bold"
  },
  [`& .${classes.cellRowLast}`]: {
    justifyContent: "space-between",
    alignItems: "center",
    display: "fex"
  },
  [`& .${classes.iteratorCellSmall}`]: {
    maxWidth: "170px",
    "& input": {
      padding: "18px 12px",
      textAlign: "center"
    }
  },
  [`& .${classes.icon}`]: {
    fill: theme.colours.inter,
    cursor: "pointer"
  }
}));
