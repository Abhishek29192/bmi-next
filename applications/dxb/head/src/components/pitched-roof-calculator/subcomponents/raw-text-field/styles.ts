import TextField from "@bmi-digital/components/text-field";
import { inputBaseClasses } from "@mui/material/InputBase";
import { styled } from "@mui/material/styles";

export const StyledTextField = styled(TextField)({
  [`& .${inputBaseClasses.input}`]: {
    zIndex: 1
  }
});
