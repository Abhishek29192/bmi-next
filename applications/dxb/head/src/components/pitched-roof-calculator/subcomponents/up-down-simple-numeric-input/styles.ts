import Button from "@bmi-digital/components/button";
import { filledInputClasses } from "@mui/material/FilledInput";
import { iconButtonClasses } from "@mui/material/IconButton";
import { styled } from "@mui/material/styles";

export const StyledComponentWithButtons = styled("div")(({ theme }) => ({
  display: "inline-flex",

  [`&& .${filledInputClasses.root}`]: {
    height: "100%",
    borderRadius: 0,
    borderRight: 0,
    borderLeft: 0,

    [`.${filledInputClasses.input}`]: {
      padding: "0 12px"
    }
  },

  "& .input": {
    margin: 0,
    display: "flex",
    flex: 1
  }
}));

export const StyledDownButton = styled(Button)(({ theme }) => ({
  borderTopLeftRadius: "3px",
  borderBottomLeftRadius: "3px",
  height: "auto",

  [`&.${iconButtonClasses.root}`]: {
    width: "56px",
    borderTopRightRadius: "0px",
    borderBottomRightRadius: "0px"
  }
}));

export const StyledUpButton = styled(Button)(({ theme }) => ({
  borderTopRightRadius: "3px",
  borderBottomRightRadius: "3px",
  height: "auto",

  [`&.${iconButtonClasses.root}`]: {
    width: "56px",
    borderTopLeftRadius: "0px",
    borderBottomLeftRadius: "0px"
  }
}));
