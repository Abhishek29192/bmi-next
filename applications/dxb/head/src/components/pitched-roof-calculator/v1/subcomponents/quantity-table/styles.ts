import { ThemeOptions } from "@bmi/components";
import { makeStyles } from "@material-ui/styles";

export const useStyles = makeStyles(
  (theme: ThemeOptions) => ({
    root: {},
    displayIfSmall: {
      display: "block",
      [theme.breakpoints!.up!("sm")]: {
        display: "none"
      }
    },

    displayIfMedium: {
      display: "none",
      [theme.breakpoints!.up!("sm")]: {
        display: "block"
      }
    },

    smallCell: {
      width: "10%"
    },

    mediumCell: {
      textAlign: "center",
      width: "17.5%"
    },

    header: {
      minWidth: "100px",
      textAlign: "center"
    },

    firstHeader: {
      minWidth: "100px"
    },

    greyBack: {
      backgroundColor: "#f7f7f7"
    },

    smallDescription: {
      margin: "10px"
    },

    packSizes: {
      width: "100px"
    },

    cellMediumRow: {
      display: "flex",
      flexDirection: "row",
      marginBottom: "10px",
      minWidth: "30%"
    },

    cellRow: {
      display: "flex",
      flexDirection: "row",
      marginBottom: "10px"
    },

    firstDescription: {
      marginLeft: "10px",
      alignSelf: "center"
    },

    cellRowLast: {
      justifyContent: "space-between",
      alignItems: "center"
    },

    icon: {
      fill: theme.colours.inter,
      cursor: "pointer"
    },

    picture: {
      height: "80px",
      width: "80px",
      mixBlendMode: "multiply"
    },

    boldText: {
      marginTop: "3px",
      marginLeft: "5px",
      fontWeight: "bold"
    },

    iteratorCellSmall: {
      maxWidth: "170px"
    },

    iteratorCellMedium: {
      minWidth: "170px"
    }
  }),
  { name: "QuantityTable" }
);
