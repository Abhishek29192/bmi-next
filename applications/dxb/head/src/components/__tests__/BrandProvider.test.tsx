import "@testing-library/jest-dom";

import React from "react";
import { render, screen } from "@testing-library/react";
import { useTheme } from "@material-ui/core/styles";
import BrandProvider, { getBrandClassName } from "../BrandProvider";

describe("BrandProvider", () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules(); // Most important - it clears the cache
    process.env = { ...OLD_ENV }; // Make a copy
    process.env.GATSBY_ENABLE_BRAND_PROVIDER = "true";
  });

  afterAll(() => {
    process.env = OLD_ENV; // Restore old environment
  });

  it("renders", () => {
    const { container } = render(<BrandProvider>Test</BrandProvider>);
    expect(container).toMatchSnapshot();
  });

  it("renders without BrandProvider", () => {
    process.env.GATSBY_ENABLE_BRAND_PROVIDER = "false";
    const { container } = render(<BrandProvider>Test</BrandProvider>);
    expect(container).toMatchSnapshot();
  });

  it("renders with brand", () => {
    const { container } = render(
      <BrandProvider brand="Braas">Test</BrandProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders without brand", () => {
    const { container } = render(<BrandProvider>Without brand</BrandProvider>);
    expect(container).toMatchSnapshot();
  });

  it("adds brand className only when brand is provided", () => {
    const { rerender } = render(<BrandProvider>Without brand</BrandProvider>);
    expect(screen.getByTestId("brand-colors-provider")).not.toHaveClass();

    rerender(<BrandProvider brand="Unknown">With brand</BrandProvider>);
    expect(screen.getByTestId("brand-colors-provider")).not.toHaveClass();

    const brand = "Braas";
    const className = getBrandClassName(brand);
    rerender(<BrandProvider brand={brand}>With brand</BrandProvider>);
    expect(screen.getByTestId("brand-colors-provider")).toHaveClass(className);
  });

  it("modifies Material UI Theme with interaction color", () => {
    const interColor = "#d10513";

    const mockGetComputedStyle = jest.spyOn(global, "getComputedStyle");
    mockGetComputedStyle.mockReturnValueOnce({
      getPropertyValue: () => interColor
    } as any);

    const TestComponent = () => {
      const theme = useTheme();
      return <div>{theme.palette.primary.main}</div>;
    };

    render(
      <BrandProvider brand="Braas">
        <TestComponent />
      </BrandProvider>
    );

    expect(screen.getByText(interColor)).toBeInTheDocument();

    mockGetComputedStyle.mockRestore();
  });
});
