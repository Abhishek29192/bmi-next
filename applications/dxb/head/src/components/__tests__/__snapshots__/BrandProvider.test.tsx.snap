// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandProvider renders 1`] = `
<div
  class=""
  data-testid="brand-colors-provider"
>
  Test
</div>
`;

exports[`BrandProvider renders with brand 1`] = `
<div
  class="BrandRed"
  data-testid="brand-colors-provider"
>
  Test
</div>
`;

exports[`BrandProvider renders without BrandProvider 1`] = `
<div
  class=""
  data-testid="brand-colors-provider"
>
  Test
</div>
`;
